### PR TITLE
Fix Next Button

### LINKED ISSUE #7: Next button buggy
Next button on downloaded questions loads next page instead of next set of downloaded questions.
image

### PR DESCRIPTION
**Pull request by Aqua-16**
## Description

I have fixed the next button by changing its behavior based on what window is opened. I have also split the downloaded questions into separate pages to make them easier to read.

## Related Issue

Fixes #7 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Motivation and Context

This solves a bug in the code and also lets the user easily see their downloaded questions.

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Aqua-16** [comment]
Been working on this since yesterday, learned a lot from this. I appreciate any feedback.
Also sir, I would love to know if there's a shorter way to implement this.

**Comment by shank03** [comment]
There are few changes required first.
- CamelCasing of variable names
- Formatting of code (Ctrl + Shift + L or Cmd + Option + L)

**Comment by shank03** [comment]
@Aqua-16 Can you give general flow of your implementation ? There are many gaps I am unable to fill.

**Comment by Aqua-16** [comment]
@shank03  Alright! 

The variable mShowingDownload is keeping track of whether the download list is showing or not. The behavior of the next and prev page buttons depends upon this.

I have also separated the makeDownloadList function from filterDownloaded since I use it independently as well.

Also, every downloaded question has an associated page key with it so that we know which page we need to show it on. 
I've also added a download_number key to the JSON file to track how many questions are downloaded.

Now, when the user logs in, the download_number is either created at a 0 value if its not present from before, or the downloadNumber variable takes the value attached with that user.

Every time the user downloads a question, the download counter is updated accordingly. By dividing the total number of downloads present at any point by a threshold (I chose 25), I have divided them into pages.

If the user logs-out, I clear their downloadNumber to 0 since all questions that were downloaded have been deleted.

**Comment by shank03** [comment]
@Aqua-16 

Right, so the `25`, make it into a constant with comments that says about the use of this.
And the new key names introduced should also be made constant in `AppData` with it's description.

**Comment by Aqua-16** [comment]
Done sir

File: src/main/java/com/shank/offcoder/cf/DownloadManager.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
 import java.util.List;
 
 public class DownloadManager {
+    private static final int pageLimit = 25; // Threshold for number of downloaded questions displayed per page.
```
Comment (shvdx) [inline src/main/java/com/shank/offcoder/cf/DownloadManager.java]: @Aqua-16 final constants are named in CAPS.

`private static final int PAGE_LIMIT = 25;`

File: src/main/java/com/shank/offcoder/controllers/Controller.java [hunk 2]
```diff
@@ -231,11 +231,11 @@ public void attemptLogin(String handle, String password, boolean auto) {
...
-                    if (AppData.get().getData("download_number", 0) == 0) {
+                    if (AppData.get().getData("AppData.DOWNLOADED_NUM", 0) == 0) {
                         AppData app = AppData.get();
-                        app.writeData("download_number", 0);
+                        app.writeData("AppData.DOWNLOADED_NUM", 0);
                     } else {
-                        downloadNumber = AppData.get().getData("download_number", 0);
+                        downloadNumber = AppData.get().getData("AppData.DOWNLOADED_NUM", 0);
                     }
                     handleField.setText("");
```
Comment (shvdx) [inline src/main/java/com/shank/offcoder/controllers/Controller.java]: Why are the constants mentioned inside double quotes ? @Aqua-16

**Review by shank03 — CHANGES_REQUESTED** [review]

File: src/main/java/com/shank/offcoder/cf/DownloadManager.java (hunk 1 above)
Comment (Aqua-16) [reply src/main/java/com/shank/offcoder/cf/DownloadManager.java]: Understood sir

File: src/main/java/com/shank/offcoder/controllers/Controller.java (hunk 2 above)
Comment (Aqua-16) [reply src/main/java/com/shank/offcoder/controllers/Controller.java]: The default didn't give me an error. I forgot to use "" in the replace. I'll fix it in a second.

**Review by shank03 — APPROVED** [review]

**Pull request merged**